### PR TITLE
Allow signed keys

### PR DESCRIPTION
**Pull request by robertkentish**
PR as per issue #479. PRoposal to support public certs for RSA signatures as in the following manner.

var keyFile = new PrivateKeyFile(@"C:\temp\ssh_keys\id_rsa", "xxxx");
var certFile = new PublicKeyCertFile(@"C:\temp\ssh_keys\id_rsa-cert.pub");
var authMethod = new PrivateKeyCertAuthenticationMethod("user", keyFile, certFile);
var connectionInfo = new ConnectionInfo("192.168.1.1", "user", authMethod);

using (var client = new SshClient(connectionInfo))
{
    client.Connect();
}

### CODE REVIEW

File: src/Renci.SshNet/Security/CertificateHostAlgorithm.cs [hunk 1]
```diff
@@ -7,12 +10,23 @@ namespace Renci.SshNet.Security
     /// </summary>
     public class CertificateHostAlgorithm : HostAlgorithm
```
Comment (drieseng) [inline src/Renci.SshNet/Security/CertificateHostAlgorithm.cs:11]: Does this class have to be public? We do not expose an instance publicly right now.
If not, **RsaCertificate** should also not be public.

File: src/Renci.SshNet/Security/CertificateHostAlgorithm.cs [hunk 2]
```diff
@@ -46,5 +84,280 @@ public override bool VerifySignature(byte[] data, byte[] signature)
...
+            /// Gets or sets the name of the algorithm as UTF-8 encoded byte array.
+            /// </summary>
+            /// <value>
+            /// The name of the algorithm.
+            /// </value>
+            private byte[] _name { get; set; }
+            public string Name
+            {
+                get { return Utf8.GetString(_name, 0, _name.Length); }
+                set { _name = Utf8.GetBytes(value); }
```
Comment (drieseng) [inline src/Renci.SshNet/Security/CertificateHostAlgorithm.cs:101]: Do any of these properties actually need a public setter?

File: src/Renci.SshNet/Security/Cryptography/RsaCertificate.cs [hunk 3]
```diff
@@ -0,0 +1,157 @@
+﻿using System;
+using System.Collections.Generic;
+using Renci.SshNet.Common;
+
+namespace Renci.SshNet.Security
+{
+    /// <summary>
+    /// 
+    /// </summary>
+    public class RsaCertificate : IDisposable
```
Comment (drieseng) [inline src/Renci.SshNet/Security/Cryptography/RsaCertificate.cs:10]: If there's no use in **CertificateHostAlgorithm** being public, we should also make this class internal.

**Comment by drieseng** [comment]
@robertkentish I did an initial sweep of your changes. Still need to dive a little deeper though.
I wonder if - when complete - we should consider shipping this as a separate package (under the **SSH.NET** unbrella, or not).

**Comment by WojciechNagorski** [comment]
@robertkentish PR needs refreshing.

**Pull request closed without merging**